### PR TITLE
CLOUDP-251510: test/helper/retry: add helper for get+mutate+update retries

### PR DESCRIPTION
**Pull request by s-urbaniak**
### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: test/int/deployment_test.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"strconv"
 	"time"
 
+	retry2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/retry"
```
Comment (josvazg) [inline test/int/deployment_test.go:11]: nitpick
may I suggest
```suggestion
	akoretry "github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/retry"
```

**Review by josvazg — APPROVED** [review]
Great!

File: test/int/deployment_test.go (hunk 1 above)
Comment (s-urbaniak) [reply test/int/deployment_test.go:11]: good catch, I chose `testretry` though as this is explicitly a test helper.

Comment (s-urbaniak) [reply test/int/deployment_test.go:11]: eh, no, `akoretry` is better 👍

**Pull request merged**